docs/modules/site-integration/nav.adoc

```diff
@@ -0,0 +1,2 @@
+* Site integration
```

Comment:
When I look at the TOC, it really feels like this should say "Maven Site Integration" instead of just "Site Integration". We may understand the context well, but for someone looking for it, I think "Site Integration" is going to be too abstract to be understood.

Reply:
2 observations:
* It gets to a point in which I have read through the content so many times that words lose meaning to me :dizzy_face:  I greatly appreaciate reviews like this.
* If it seems Site sections is "shoe horned", it is. There's not enough content to make several pages, but I wanted to have a two level toc for consistency.

This is the final toc, I capitalized all titles for consistency

![image](https://user-images.githubusercontent.com/5781153/100284835-23e53a80-2f70-11eb-8771-4d0e35f431a2.png)